### ISSUE TITLE
Document GPS fields for 3.10 and before

### ISSUE THREAD

**Issue by mrjones-plip** [medic/cht-docs#326]
While working in [config-moh-mali#89](https://github.com/medic/config-moh-mali/issues/89) (private repo), we've come across a number of issues trying to decipher where the canonical GPS values are in a given document. We did a bunch of work in [3.10](https://github.com/medic/cht-core/blob/master/release-notes/docs/3.10.0.md) to [improve GPS](https://github.com/medic/cht-core/issues/5917) in cht-core - we should document these fields!

For example, in a given form we have the JSON below the lat of "1" and lon of "2" were manually entered and stored in `fields.location.gps_ask`: `1 2 3 4`.  But why is `hidden_fields.location.gps_ask` and `fields.inputs.meta.location` empty?  Why does `geolocation.latitude/longitude` have values from a prior GPS reading not applicable to this document? (see [this comment](https://github.com/medic/config-moh-mali/issues/89#issuecomment-700352652) for details on how this last item was achieved).

Just as important, and clearly releated,  we should try and document where you get the correct values when dealing with different versions of cht-core. This query from @ecsalomon in [this .sql file](https://github.com/medic/medic-impact/blob/master/utils/sql_recipe_book/all_gps_records.sql) (private repo):

```sql
COALESCE(
  NULLIF(doc #>> '{fields,inputs,meta,location,lat}', ''),
  NULLIF(doc #>> '{fields,geolocation,latitude}', ''),
  NULLIF(doc #>> '{geolocation,latitude}', '')
)  AS location_lat,
```

This tries to account for all the places cht-core might have the latitude (and hasn't been updated for JSON locations which may have changed in 3.10).

----
```json
{
  "_id": "615919d3c437",
  "_rev": "1-bb8037fde933",
  "form": "assessment",
  "type": "data_record",
  "content_type": "xml",
  "reported_date": 1601335954857,
  "contact": {
    "_id": "fa818e9bb77b",
    "parent": {
      "_id": "6069980b6b88b",
      "parent": {
        "_id": "1b86a5d342297",
        "parent": {
          "_id": "2c36b40a13a7c"
        }
      }
    }
  },
  "from": "",
  "hidden_fields": [
    "place_id",
    "location.region_label",
    "location.district_label",
    "location.region_show",
    "location.district_show",
    "location.gps_ask",
    "presentation.n_introduce",
    "presentation.n_explain",
    "thanks",
    "meta"
  ],
  "fields": {
    "inputs": {
      "meta": {
        "location": {
          "lat": "",
          "long": "",
          "error": "",
          "message": ""
        },
        "deprecatedID": ""
      },
      "source": "user",
      "source_id": "",
      "user": {
        "contact_id": "fa89bb77b",
        "name": "user-name",
        "editing": "no",
        "role": "chw",
        "parent": {
          "_id": "6069380b6b88b",
          "region": "user_region",
          "district": "user-district",
          "parent": {
            "_id": "1b842297",
            "parent": {
              "_id": "2c36b778-ab4a-2d1140a13a7c"
            }
          }
        }
      },
      "contact": {
        "_id": "",
        "name": "",
        "parent": {
          "_id": "",
          "parent": {
            "_id": ""
          }
        }
      }
    },
    "place_id": "e6c3-4615-8499-967980b6b88b",
    "location": {
      "region_label": "region_label_here",
      "district_label": "district_label_here",
      "region_name": "region_name",
      "district_name": "district_name",
      "region_show": "region_show",
      "district_show": "district_show",
      "village": "other",
      "village_text": "other2",
      "gps_ask": "1 2 3 4",
      "gps": "1 2 3 4"
    },
    "presentation": {
      "n_introduce": "",
      "n_explain": "",
      "consent": "no"
    },
    "g_repeat_count": "",
    "g_repeat_children_count": "",
    "thanks": {
      "n_thank": ""
    },
    "meta": {
      "instanceID": "uuid:4deb-90d4-f910ba082099"
    }
  },
  "geolocation": {
    "latitude": 5.3951438,
    "longitude": -15.1982217,
    "altitude": 724,
    "accuracy": 8,
    "altitudeAccuracy": null,
    "heading": 93,
    "speed": 2.2002954483032227
  },
  "_attachments": {
    "content": {
      "content_type": "application/xml",
      "revpos": 1,
      "digest": "md5-SwW5T/wivqUobWaFmtfDeA==",
      "length": 2564,
      "stub": true
    }
  }
}
```

**Comment by mrjones-plip** [comment]
fyi @dianabarsan  @binokaryg @sacul-git @Marina-Kenf

**Comment by binokaryg** [comment]
Thanks @mrjones-plip. As far as I know, the fields inside `hidden_fields` are always empty and this is by design. It is the list of fields that we configure to not show in the app's report section as they are less relevant to the user.

**Comment by MaxDiz** [comment]
consolidating from issue #296 

> It's an important enough topic that we might also want to add some content to the Concepts or Features section
> 
> Note to author: The app will wait up to 30 seconds to get a location lock before it gives up. If a user tries to submit a form before a location lock is obtained and before 30 seconds are up, the submit action will wait until 30 seconds are up. Also: This change also supports the latest updates in medic-android, where the Location permission is requested at runtime, when editing/creating a report, rather than demanded when starting the app for the first time.

**Comment by mrjones-plip** [comment]
> The app will wait up to 30 seconds to get a location lock before it gives up.

@MaxDiz  this only applies to >3.10, correct?

**Comment by MaxDiz** [comment]
correct

Max Dizard | Product Manager

Medic Mobile <https://medicmobile.org/> | @Medic <https://twitter.com/medic>
| Facebook <https://www.facebook.com/MedicMobile>


Join our community. Sign up for Medic Mobile's newsletter.
<https://us8.list-manage.com/subscribe?u=e5b07d2afb4b273eaff8e3c53&id=6f368ee7ca>

**Comment by mrjones-plip** [comment]
Cool cool! With the request to document earlier versions, I wanted to be sure the author of the new docs delineated which version supported which GSP features. Thanks for confirming!

**Comment by mrjones-plip** [comment]
Also for the author: turns out that `<location><error/><message/></location>` being added to forms will enable error collection [per cht-core#2670](https://github.com/medic/cht-core/issues/2670#issuecomment-243429644). This should be added to *all* GPS fields as far as I can tell.  Be sure to reference [GeolocationPositionError](https://developer.mozilla.org/en-US/docs/Web/API/GeolocationPositionError) for what gets put in `error` and `message` fields.

**Comment by andrablaj** [comment]
@mrjones-plip as 3.10 is not supported anymore, is this ticket still worth the effort?

**Comment by mrjones-plip** [comment]
yeah - I can go with that.  It has good references in here that are text (and this searchable/findable) - we can re-open if need be!
